src/components/recordedit/recordedit.tsx

```diff
@@ -350,56 +447,258 @@ const RecordeditInner = ({
+  const onSelectedRowsChanged = (rows: SelectedRow[]) => {
+    // return "false" to disable submit button in modal
+    let numForms = forms.length;
+
+    // if we fill the first form, then reduce our calculation by 1
+    // NOTE: forms.length "should" be 1 at this point before subtracting 1
+    if (selectionsFillFirstForm) numForms--;
+
+    return (rows.length + numForms < RECORDEDIT_MAX_ROWS);
+  }
+
+  // user closes the modal without making any selections
+  const closeBulkForeignKeyCB = () => {
+    // if the page was loaded with a modal showing and it is dismissed, update app state variable and do nothing else
+    if (selectionsFillFirstForm) setSelectionsFillFirstForm(false);
+
+    setShowBulkForeignKeyModal(false);
+  }
+
+  /**
+   * user makes selections in the multi select foreign key modal and clicks submit
+   * this function updates the selected rows (if the foreign keys are part of a unique key) and fills in the new forms based
+   * on the state of the app and the number of selected rows
+   *
+   * if the first modal is submitted after load of app page, one of the selected values will
+   * fill in the first form. After that, the selections will copy the last form's values or use default
+   * values based on what is set in the annotation (pending annotation implementation)
+   *
+   * NOTE: This should only be called if reference.bulkCreateForeignKeyObject is defined
+   *
+   * @param modalSelectedRows the selected rows from the foreign key modal
+   */
+  const submitBulkForeignKeyCB = (modalSelectedRows: SelectedRow[]) => {
+    setShowBulkForeignKeyModal(false);
+
+    // should not happen since submit button is greyed out
+    if (!modalSelectedRows || modalSelectedRows.length === 0) return;
+
+    const bulkFKObject = reference.bulkCreateForeignKeyObject
+    if (bulkFKObject.isUnique) {
+      /**
+       * copy modalSelectedRows 2nd to preserve indexes in bulkForeignKeySelectedRows
+       *
+       * this function does 2 different things:
+       *  - fills the first form and adds new forms
+       *  - OR only adds new forms
+       *
+       * in both cases, the selected rows are added to the forms in the same order that
+       * the rows were selected in the modal. As we are adding new forms, we copy the
+       * values from the modalSelectedRows in the same index order
+       **/
+      const newRows = [...bulkForeignKeySelectedRows, ...modalSelectedRows]
+      setbulkForeignKeySelectedRows(newRows);
+    }
+
+    // recordedit has already been initialized so start adding new forms
+    const tempFormValues = methods.getValues();
+    let initialValues = tempFormValues,
+      startFormNumber: number;
+
+    if (selectionsFillFirstForm) {
+      if (modalSelectedRows.length > 1) {
+        initialValues = createNewForms(tempFormValues, modalSelectedRows.length - 1).tempFormValues;
+      }
+
+      startFormNumber = 1;
+
+      setSelectionsFillFirstForm(false);
+      // } else if (annotation.doClone) {
```

Comment:
I don't think we're going to allow this anymore, right? So you should just remove the commented code.